### PR TITLE
[APB-4586][SM]: Migrate ASA service to ACA

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationFastTrackJourneyController.scala [hunk 1]
```diff
@@ -48,9 +48,9 @@ import scala.concurrent.ExecutionContext
 @Singleton
 class AgentInvitationFastTrackJourneyController @Inject()(
   invitationsService: InvitationsService,
-  invitationsConnector: InvitationsConnector,
-  asaConnector: AgentServicesAccountConnector,
+  invitationsConnector: AgentClientAuthorisationConnector,
   relationshipsService: RelationshipsService,
+  acaConnector: AgentClientAuthorisationConnector,
```
Comment (mariannepearson0) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationFastTrackJourneyController.scala]: injecting AgentClientAuthorisationConnector twice with 2 different names

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationFastTrackJourneyController.scala [hunk 2]
```diff
@@ -132,8 +132,8 @@ class AgentInvitationFastTrackJourneyController @Inject()(
   val submitCheckDetails = action { implicit request =>
     whenAuthorisedWithForm(AsAgent)(checkDetailsForm)(
       Transitions.checkedDetailsAllInformation(checkPostcodeMatches)(checkCitizenRecordMatches)(
-        checkVatRegistrationDateMatches)(createInvitation)(createAgentLink)(getAgencyEmail)(hasPendingInvitationsFor)(
-        hasActiveRelationshipFor))
+        checkVatRegistrationDateMatches)(invitationsService.createInvitation)(invitationsService.createAgentLink)(
```
Comment (mariannepearson0) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationFastTrackJourneyController.scala]: inivtationsService is being imported above on line 86 so don't need the invitationsService. prefix

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationJourneyController.scala [hunk 3]
```diff
@@ -43,9 +43,9 @@ import scala.concurrent.ExecutionContext
 @Singleton
 class AgentInvitationJourneyController @Inject()(
   invitationsService: InvitationsService,
-  invitationsConnector: InvitationsConnector,
+  invitationsConnector: AgentClientAuthorisationConnector,
   relationshipsService: RelationshipsService,
-  asaConnector: AgentServicesAccountConnector,
+  acaConnector: AgentClientAuthorisationConnector,
```
Comment (mariannepearson0) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationJourneyController.scala]: again AgentClientAuthorisationConnector is being injected twice

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsRequestTrackingController.scala [hunk 4]
```diff
@@ -55,10 +55,10 @@ class AgentsRequestTrackingController @Inject()(
   val featureFlags: FeatureFlags,
   val trackService: TrackService,
   val invitationsService: InvitationsService,
-  val invitationsConnector: InvitationsConnector,
+  val invitationsConnector: AgentClientAuthorisationConnector,
   val relationshipsConnector: RelationshipsConnector,
   val pirRelationshipConnector: PirRelationshipConnector,
-  agentServicesAccountConnector: AgentServicesAccountConnector,
+  acaConnector: AgentClientAuthorisationConnector,
```
Comment (mariannepearson0) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsRequestTrackingController.scala]: and here the connector is being injected twice

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationFastTrackJourneyController.scala (hunk 2 above)
Comment (sureshhmrc) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationFastTrackJourneyController.scala]: we have same method name in both inivtationsService and acaConnector - so needed this prefix

Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationFastTrackJourneyController.scala]: 👍

**Pull request merged**